programs/whirlpool/src/instructions/v2/set_config_extension_authority.rs

```diff
@@ -0,0 +1,25 @@
+use anchor_lang::prelude::*;
+
+use crate::state::{WhirlpoolsConfig, WhirlpoolsConfigExtension};
+
+#[derive(Accounts)]
+pub struct SetConfigExtensionAuthority<'info> {
+    pub whirlpools_config: Box<Account<'info, WhirlpoolsConfig>>,
+
+    #[account(mut, has_one = whirlpools_config)]
+    pub whirlpools_config_extension: Account<'info, WhirlpoolsConfigExtension>,
+
+    #[account(address = whirlpools_config_extension.config_extension_authority)]
+    pub config_extension_authority: Signer<'info>,
+
+    /// CHECK: safe, the account that will be new authority can be arbitrary
+    pub new_config_extension_authority: UncheckedAccount<'info>,
```

Comment:
TokenBadge and WhirlpoolsConfigExtension are not "_v2" instruction.

They are included in /v2/ because they are implemented for TokenExtension.
If you feel uncomfortable with the location of the source code, we can change it.

In addition, I would like to add that I have separated the dir/file to make it as clear as possible that the v1 code has not been changed. (Not destroying v1 is an important point for this change)

Reply:
Yep think the current setup makes sense to leave v1 as is